### ISSUE TITLE
[BUG] Calling Get-PnPFile with -ThrowExceptionIfFileNotFound makes return type ListItem without -AsListItem

### ISSUE THREAD

**Issue by pbjorklund** [pnp/powershell#209]
### Reporting an Issue or Missing Feature

```Get-PnPFile -Url $file.SourceFileLocation -Connection $sourceConn -ErrorAction Stop```
Will return Microsoft.SharePoint.Client.File

```Get-PnPFile -Url $file.SourceFileLocation -ThrowExceptionIfFileNotFound -Connection $sourceConn -ErrorAction Stop```
Will return Microsoft.SharePoint.Client.ListItem

Documentation states:

_-ThrowExceptionIfFileNotFound
If provided in combination with -AsListItem, a System.ArgumentException will be thrown if the file specified in the -Url argument does not exist. Otherwise it will return nothing instead._

Seems that calling with ThrowExceptionIfFileNotFound actually is an implicit -AsListItem. 

Maybe a warning might be in place? It was not obvious to me why it didn't give me a File.

### What is the version of the Cmdlet module you are running?
1.2.4      nightly

### Which operating system/environment are you running PnP PowerShell on?
- [x] Windows
- [ ] Linux
- [ ] MacOS
- [ ] Azure Cloud Shell
- [ ] Azure Functions
- [ ] Other : please specify

**Comment by erwinvanhunen** [comment]
The parameter was introduced later to throw an exception but only in the case of retrieving a list item. The other switches already threw an error, just not the -AsListItem. To not break the experience we added an explicit flag for this in that specific parameter set. However, as we made the main switches (-AsString, -AsListItem etc.) not required within the parameter set it would already resolve to the parameter set associated with the switch.

Long story short: it has been fix and will be available in 1.2.13-nightly.
